domain/global_vars_cache.go

```diff
@@ -67,24 +67,22 @@ func (gvc *GlobalVariableCache) Disable() {
 	return
 }
 
-// checkEnableStmtSummary looks for TiDBEnableStmtSummary and notifies StmtSummary
-func checkEnableStmtSummary(rows []chunk.Row, fields []*ast.ResultField) {
+// checkEnableServerGlobalVar processes variables that acts in server and global level.
+func checkEnableServerGlobalVar(rows []chunk.Row) {
 	for _, row := range rows {
-		varName := row.GetString(0)
-		if varName == variable.TiDBEnableStmtSummary {
-			varVal := row.GetDatum(1, &fields[1].Column.FieldType)
-
+		switch row.GetString(0) {
+		case variable.TiDBEnableStmtSummary:
 			sVal := ""
-			if !varVal.IsNull() {
-				var err error
-				sVal, err = varVal.ToString()
-				if err != nil {
-					return
-				}
+			if !row.IsNull(1) {
+				sVal = row.GetString(1)
 			}
-
 			stmtsummary.StmtSummaryByDigestMap.SetEnabled(sVal, false)
-			break
+		case variable.TiDBCapturePlanBaseline:
```

Comment:
Shall we add dependency check between these 2 variables?  if not, would there be possible panics when accessing structures of statements?

Reply:
No, it won't panic because the `StmtSummaryByDigestMap` is initilized as a global variable, so it is always valid.